WordPress/Classes/Utility/Networking/RequestAuthenticator.swift

```diff
@@ -221,6 +225,46 @@ class RequestAuthenticator: NSObject {
         }
     }
 
+    private func requestForMappedWPCom(url: URL, cookieJar: CookieJar, username: String, authToken: String, siteID: Int, completion: @escaping (URLRequest) -> Void) {
+        func done() {
+            guard
+                let host = url.host,
+                !host.contains("wordpress.com")
+            else {
+                // The requested URL is to the unmapped version of the domain,
+                // so skip proxying the request through r-login.
+                let request = URLRequest(url: url)
+                completion(request)
+                return
+            }
+
+            let rlogin = "https://r-login.wordpress.com/remote-login.php?action=auth"
+            var components = URLComponents(string: rlogin)
```

Comment:
I'd suggest to deal with this optional early, since it'd signal an abnormal situation.  I'd also log an error here, so that we have a better chance of fixing whatever issue there is.

```swift
guard let components = URLComponents(string: rlogin) else {
    DDLogError("There's something wrong with the rlogin URL: \(login).")
    completion(URLRequest(url: url))
    return
}

// The rest of the code below doesn't need to deal with the optional
```